### PR TITLE
Fix closing of modal window curtains while dragging and resizing

### LINKED ISSUE #7496: Modal window draggingCurtain lingers when closed while dragging
**Originally by _sebgymn_**

___
A modal window places a div with class `v-window-draggingCurtain` that effectively disallows user to click the background. However, in certain cases, after the window is closed, this element does not go away, rendering the whole application unusable. 

STR: 

1. Bind escape button (I have not tried other buttons) for closing the window: `window.addCloseShortcut(ShortcutAction.KeyCode.ESCAPE)`
2. On the screen, while dragging the window from its title (while the mouse button is still down), hit escape button to close the window.
3. The draggingCurtain does not go away and you cannot interact with the application (until you delete that element from the browser inspector).

This can also be observed in https://demo.vaadin.com/valo-theme/#!common 
Open a closable modal window and apply step 2.

Thank you.

___
 Imported from https://dev.vaadin.com/ issue #19574

### PR DESCRIPTION
**Pull request by ahie**
Fixes #7496

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/framework/8281)

### CODE REVIEW

File: uitest/src/test/java/com/vaadin/tests/themes/valo/ModalWindowTest.java [hunk 1]
```diff
@@ -30,7 +34,30 @@ public void modalAnimationsAreDisabled() {
...
     }
 
+    @Test
+    public void modal_curtains_close_correctly() {
+        openTestURL();
+
+        openModalWindow();
+        new Actions(getDriver())
+                .moveToElement(findElement(By.className("v-window-header")))
+                .clickAndHold().sendKeys(Keys.ESCAPE).perform();
```
Comment (tsuoanttila) [inline uitest/src/test/java/com/vaadin/tests/themes/valo/ModalWindowTest.java]: I'd like to see an extra validation step in between starting the drag and pressing escape to see that the curtain is present. Some Drag and Drop cases in the framework require a movement on the mouse before any drag functionality is started.

Comment (ahie) [inline uitest/src/test/java/com/vaadin/tests/themes/valo/ModalWindowTest.java]: Done.

**Comment by tsuoanttila** [comment]
Started a full test build. If no issues are discovered :lgtm:

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8281#-:-KaqCnaNiYOB0zRvFoP9:b-v3qtu)*

**Comment by tsuoanttila** [comment]
Tests passed, update branch and merge.

**Review by tsuoanttila — APPROVED** [review]

**Pull request merged**